plugins/outputs/azure_data_explorer/azure_data_explorer.go

```diff
@@ -0,0 +1,200 @@
+}
+
+type ingestorFactory func(localClient, string, string) (localIngestor, error)
+
+const createTableCommand = `.create-merge table ['%s']  (['fields']:dynamic, ['name']:string, ['tags']:dynamic, ['timestamp']:datetime);`
+const createTableMappingCommand = `.create-or-alter table ['%s'] ingestion json mapping '%s_mapping' '[{"column":"fields", "Properties":{"Path":"$[\'fields\']"}},{"column":"name", "Properties":{"Path":"$[\'name\']"}},{"column":"tags", "Properties":{"Path":"$[\'tags\']"}},{"column":"timestamp", "Properties":{"Path":"$[\'timestamp\']"}}]'`
+
+func (adx *AzureDataExplorer) Description() string {
+	return "Sends metrics to Azure Data Explorer"
+}
+
+func (adx *AzureDataExplorer) SampleConfig() string {
+	return `
+  ## Azure Data Exlorer cluster endpoint
+  ## ex: endpoint_url = "https://clustername.australiasoutheast.kusto.windows.net"
+  endpoint_url = ""
+  
+  ## The Azure Data Explorer database that the metrics will be ingested into.
+  ## The plugin will NOT generate this database automatically, it's expected that this database already exists before ingestion.
+  ## ex: "exampledatabase"
+  database = ""
+
+  ## Client ID of the Azure Active Directory App (Service Principal). This Service Principal should have permissions on the Azure Data Explorer database
+  ## to create Tables and ingest data into these tables
+  ## ex: client_id = "dc871111-1222-4eee-bwww-111111111111"
+  client_id = ""
+
+  ## The client secret of the Service Principal in Azure that has ingestion rights to the Azure Data Exploer Cluster
+  client_secret = ""
+
+  ## The tenant ID of the Azure Subsciption in which the Service Principal belongs to
+  tenant_id = ""
+
+  ## Timeout for Azure Data Explorer operations
+  # timeout = "15s"
+`
+}
+
+func (adx *AzureDataExplorer) Connect() error {
+	authorizer := kusto.Authorization{
+		Config: auth.NewClientCredentialsConfig(adx.ClientID, adx.ClientSecret, adx.TenantID),
+	}
+
+	client, err := kusto.New(adx.Endpoint, authorizer)
+
+	if err != nil {
+		return err
+	}
+	adx.client = client
+	adx.ingesters = make(map[string]localIngestor)
+	adx.createIngestor = createRealIngestor
+
+	return nil
+}
+
+func (adx *AzureDataExplorer) Close() error {
+	adx.client = nil
+	adx.ingesters = nil
+
+	return nil
+}
+
+func (adx *AzureDataExplorer) Write(metrics []telegraf.Metric) error {
+	metricsPerNamespace := make(map[string][]byte)
+	// Group metrics by name and serialize them
+	for _, m := range metrics {
+		namespace := m.Name()
+		metricInBytes, err := adx.serializer.Serialize(m)
+		if err != nil {
+			return err
+		}
+		if existingBytes, ok := metricsPerNamespace[namespace]; ok {
+			metricsPerNamespace[namespace] = append(existingBytes, metricInBytes...)
+		} else {
+			metricsPerNamespace[namespace] = metricInBytes
+		}
+	}
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(adx.Timeout))
```

Comment:
should this timeout be longer if there's a lot of namespaces? seems likely it could not submit in time if there's a lot of unique measurements

Reply:
Just increased the default timeout to 20 seconds according to your comment. In all cases, this is a config value and the user should be able to update it if necessary.